src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,51 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
 
-    public MergeSortAction(int[] array) {
+public class MergeSortAction extends RecursiveAction {
+    private int[] arr;
 
+    public MergeSortAction(int[] arr) {
+        this.arr = arr;
+    }
+
+    @Override
+    protected void compute() {
+        if (arr.length < 2) {
+            return;
+        }
+        var left = Arrays.copyOfRange(arr, 0, arr.length / 2);
+        var right = Arrays.copyOfRange(arr, arr.length / 2, arr.length);
```

Comment:
better use System.arraycopy

Reply:
System.arraycopy has void return type, so I also need to create two separate arrays. And as I found Arrays.copyOfRange uses System.arraycopy under the hood :)